pgjdbc/src/main/java/org/postgresql/core/v3/QueryExecutorImpl.java

```diff
@@ -2912,26 +2912,78 @@ public ReplicationProtocol getReplicationProtocol() {
     return replicationProtocol;
   }
 
+  @Override
+  public void addBinaryReceiveOid(int oid) {
+    synchronized (useBinaryReceiveForOids) {
```

Comment:
Do we really want synchronized in here ? I thought we were trying to remove sync ?

Reply:
We do need to synchronize, since `HashMap` is not thread safe.

Reply:
I figured that we needed to. I just thought since we are going to be making changes for loom we might want to use a loom compatible mechanism. The time in the `HashMap` is pretty short so my point is pretty moot